src/resource-builder.ts

```diff
@@ -1,14 +1,15 @@
 import { Resource } from '@opentelemetry/resources';
+import { HoneycombOptions } from './honeycomb-options';
 import { VERSION } from './version';
 
 /**
  * Builds and returns an Opentelemetry Resource with
  * added resource attributes specific to the Honeycomb Distro
  * @returns a Resource instance
  */
-export function configureHoneycombResource(): Resource {
+export function configureHoneycombResource(opts?: HoneycombOptions): Resource {
   return new Resource({
     'honeycomb.distro.version': VERSION,
     'honeycomb.distro.runtime_version': process.versions.node,
-  });
+  }).merge(opts?.resource ? opts.resource : null);
```

Comment:
This looks awful but I wanted to use the `merge` function which either takes a `Resource` type or `null` 😭